hoomd/hpmc/IntegratorHPMC.h

```diff
@@ -437,6 +438,100 @@ class PYBIND11_EXPORT IntegratorHPMC : public Integrator
         return m_patch;
         }
 
+    /// Test if this has pairwise interactions.
+    bool hasPairInteractions()
+        {
+        return m_patch || m_pair_potentials.size() > 0;
+        }
+
+    /// Get pairwise interaction maximum non-additive r_cut.
+    LongReal getMaxPairEnergyRCutNonAdditive() const
+        {
+        LongReal r_cut = 0;
+        if (m_patch)
+            {
+            r_cut = m_patch->getRCut();
+            }
+        for (const auto& pair : m_pair_potentials)
+            {
+            for (unsigned int type_i = 0; type_i < m_pdata->getNTypes(); type_i++)
+                {
+                for (unsigned int type_j = 0; type_j < m_pdata->getNTypes(); type_j++)
+                    {
+                    r_cut = std::max(r_cut, pair->getRCutNonAdditive(type_i, type_j));
```

Comment:
Maybe it makes sense to have a `PairPotential::getMaxRCutNonAdditive` method to just do a linear search over half of the `std::vector`?

Reply:
Yes. And/or have PairPotential cache the maximimum or move all calls to `getMaxRCutNonAdditive` outside of loops so it is called once/step instead of once/particle.